languages/js/src/Query.ts

```diff
@@ -50,9 +50,10 @@ export class Query {
     this.#calls = new Map();
     this.#host = host;
 
-    for (const k in bindings) {
-      this.bind(k, bindings.get(k));
-    }
+    if (bindings)
+      for (const [k, v] of bindings) {
+        this.bind(k, v);
+      }
```

Comment:
This was the source of the bug. The loop wasn't actually iterating over anything, so the initial type constraint on `resource` wasn't being set, so it was matching a rule it shouldn't have matched which was inducing an impossible constraint.

Reply:
WOW